catroid/src/main/java/org/catrobat/catroid/backpack/BackpackLookController.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.backpack;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.FileChecksumContainer;
+import org.catrobat.catroid.common.LookData;
+import org.catrobat.catroid.io.StorageHandler;
+import org.catrobat.catroid.ui.controller.BackPackListManager;
+import org.catrobat.catroid.utils.Utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public final class BackpackLookController {
+
+	public static LookData pack(LookData item) throws IOException {
+		List<LookData> backpackedLooks = BackPackListManager.getBackPackedLooks();
+		if(backpackedLooks.contains(item)) {
+			return backpackedLooks.get(backpackedLooks.indexOf(item));
+		}
+
+		File packedLookFile = getPackedLookFile(item);
+
+		LookData packedLookData = new LookData(item.getLookName(), packedLookFile.getName());
+		packedLookData.isBackpackLookData = true;
+		return packedLookData;
+
+	}
+
+	public static LookData packAndAddToVisibleBackpack(LookData item) throws IOException {
+		LookData packedLookData = pack(item);
+		BackPackListManager.addLookToBackPack(packedLookData);
+		return packedLookData;
+	}
+
+	private static File getPackedLookFile(LookData item) throws IOException {
```

Comment:
we should rename this, as the name might suggest that it returns a file which is already withing the backpack directory.

Or remove it entirely as it is only a one liner